src/runtime_src/core/common/drv/kds_core.c

```diff
@@ -159,6 +195,39 @@ kds_cu_config(struct kds_cu_mgmt *cu_mgmt, struct kds_command *xcmd)
 	return ret;
 }
 
+static int
+kds_scu_config(struct kds_scu_mgmt *scu_mgmt, struct kds_command *xcmd)
+{
+	struct ert_configure_sk_cmd *scmd;
+	int i, j;
+
+	scmd = (struct ert_configure_sk_cmd *)xcmd->execbuf;
+	for (i = 0; i < scmd->num_image; i++) {
+		struct config_sk_image *cp = &scmd->image[i];
+
+		for (j = 0; j < cp->num_cus; j++) {
+			int scu_idx = j + cp->start_cuidx;
+
+			/* TODO: Why continue? */
```

Comment:
Historical reason. Config PS kernel ert command was not atomic. There could be multiple configs from different processes. So we continue if the name has been cached. But I believe it can be removed now.

Reply:
Ok. I would remove this in next commit which send PS kernel config command to ERT in xclbin download. This can make sure it is atomic.